### PR TITLE
Bug-fix/v1 158

### PR DESCRIPTION
**Pull request by SaXaPhonist**
- fixed problem with skills sorting
- fixed problem with duplicated api request
-  fixed problem with "Loader" in skills
- fixed problem with "No skills" when searching for non-existed skill

### CODE REVIEW

File: frontend/src/api/skills/getAllSkills.ts [hunk 1]
```diff
@@ -8,16 +8,17 @@ import { errorSnackbar } from 'constants/snackbarVariant';
...
     enqueueSnackbar(error?.response?.data, errorSnackbar);
   };
   return useQuery(
-    [QUERY_KEYS.skills],
+    [QUERY_KEYS.skills, skillName],
     async () => {
       const apiClient = apiClientWrapper();
-      const response = await apiClient.get(API.skills);
+      const skillNameEncoded = encodeURIComponent(skillName);
+      const response = await apiClient.get(`${API.skills}?search=${skillNameEncoded}`);
```
Comment (ostapdans99) [inline frontend/src/api/skills/getAllSkills.ts:21]: maybe better to forward queries like params. https://masteringjs.io/tutorials/axios/get-query-params

File: frontend/src/components/SkillGroup/SkillGroup.tsx [hunk 2]
```diff
@@ -18,9 +19,11 @@ const SkillGroup: FC<ISkillGroupProps> = ({ skillFounded, skills }) => {
...
-};
+      ))
+    ) : (
+      <NoContent message={NO_SKILLS} />
+    )}
+  </>
+);
 
-export default SkillGroup;
+export default memo(SkillGroup);
```
Comment (ostapdans99) [inline frontend/src/components/SkillGroup/SkillGroup.tsx:29]: Are you shure that it works correctly? skillFounded its array.  https://blog.bitsrc.io/optimize-your-react-app-with-react-memo-ec52447b09ba. 
image

File: frontend/src/pages/Skills/SkillsContainer.tsx [hunk 3]
```diff
@@ -1,46 +1,30 @@
...
-    } else {
-      setSkillFounded([]);
-    }
-  }, [debouncedSearchValue, searchInputValue, skillSearchResponse]);
+    setSkillFounded(skillsResponse);
+  }, [skillsResponse]);
 
   const handleSearchInputChange = ({ target }: BaseSyntheticEvent) => {
-    setSearchInputValue(formatInputValue(target.value));
+    setSearchInputValue(() => formatInputValue(target.value));
```
Comment (ostapdans99) [inline frontend/src/pages/Skills/SkillsContainer.tsx:20]: is it dont the same as  setSearchInputValue(formatInputValue(target.value)); ?

Comment (SaXaPhonist) [reply frontend/src/pages/Skills/SkillsContainer.tsx:20]: yes

File: frontend/src/components/SkillGroup/SkillGroup.tsx (hunk 2 above)
Comment (SaXaPhonist) [reply frontend/src/components/SkillGroup/SkillGroup.tsx:29]: need to discuss

File: frontend/src/api/skills/getAllSkills.ts (hunk 1 above)
Comment (SaXaPhonist) [reply frontend/src/api/skills/getAllSkills.ts:21]: yes

File: frontend/src/components/SkillGroup/SkillGroup.tsx [hunk 4]
```diff
@@ -1,14 +1,15 @@
-import { FC } from 'react';
+import { FC, memo } from 'react';
 
+import NoContent from 'components/NoContent';
+import { NO_SKILLS } from 'constants/messages';
 import SkillItem from './SkillItem';
 import { ISkillGroupProps } from './types';
```
Comment (irina-sinitsyna) [inline frontend/src/components/SkillGroup/SkillGroup.tsx:6]: fix the order please

File: frontend/src/pages/Skills/Skills.tsx [hunk 5]
```diff
@@ -1,12 +1,10 @@
...
 
-import { NO_SKILLS } from 'constants/messages';
-import Loader from 'components/Loader';
 import PageTitle from 'components/PageTitle';
 import SkillGroup from 'components/SkillGroup';
-import NoContent from 'components/NoContent';
-import { Loaders } from 'enums/loader';
+import Loader from 'components/Loader';
 
+import { Loaders } from 'enums/loader';
```
Comment (irina-sinitsyna) [inline frontend/src/pages/Skills/Skills.tsx:7]: fix imports please

File: frontend/src/pages/Skills/types/index.ts [hunk 6]
```diff
@@ -3,9 +3,7 @@ import { BaseSyntheticEvent } from 'react';
 import { ISkillsListProps } from 'types/skill';
 
 export interface ISkillsPageProps {
-  skillFounded: ISkillsListProps[];
-  searchInputValue: string;
+  skillFounded: ISkillsListProps[] | undefined;
```
Comment (irina-sinitsyna) [inline frontend/src/pages/Skills/types/index.ts:6]:
```suggestion
  skillFounded?: ISkillsListProps[];
```

File: frontend/src/pages/Skills/types/index.ts [hunk 7]
```diff
@@ -3,9 +3,7 @@ import { BaseSyntheticEvent } from 'react';
...
   handleSearchInputChange: (event: BaseSyntheticEvent) => void;
-  skills?: ISkillsListProps[];
-  isLoading?: boolean;
+  isSkillsLoading: boolean;
```
Comment (irina-sinitsyna) [inline frontend/src/pages/Skills/types/index.ts:8]: fix order please

**Review by ostapdans99 — APPROVED** [review]

**Review by irina-sinitsyna — DISMISSED** [review]

**Review by irina-sinitsyna — APPROVED** [review]

File: frontend/src/types/user.ts [hunk 8]
```diff
@@ -1,6 +1,6 @@
-import { INotification } from 'types/notification';
 import { TGroupedSkills, Technologies } from './skill';
 import { IClientCourse } from './clientCourse';
+import { INotification } from './notifications';
```
Comment (k-krasilnikova) [inline frontend/src/types/user.ts:3]: Do we really need to rename this file? In other cases we're using singular form

**Pull request merged**